test/adapters/CharmAdapter.js

```diff
@@ -0,0 +1,734 @@
+const { assert, expect } = require("chai");
```

Comment:
General comments for the test file:

* Please use [Prettier](https://prettier.io/) in your IDE to format JS code
* There are references to otokens here and there, let's change them to be specific to charm